gammapy/stats/variability.py

```diff
@@ -228,3 +230,117 @@ def compute_flux_doubling(flux, flux_err, coords, axis=0):
     }
 
     return doubling_dict
+
+
+def TimmerKonig_lightcurve_simulator(
+    power_spectrum,
+    npoints,
+    spacing,
+    t_ref=None,
+    random_state="random-seed",
+    power_spectrum_params=None,
+):
+    """Implementation of the Timmer-Koenig algorithm to simulate a time series from a power spectrum.
+
+    Parameters
+    ----------
+    power_spectrum : function
+        Power spectrum used to generate the time series. It is expected to be
+        a function mapping the input frequencies to the periodogram.
+    npoints : float
+        Number of points in the output time series.
+    spacing : '~astropy.Quantity'
+        Sample spacing, inverse of the sampling rate.
+    t_ref : '~astropy.Time', optional
```

Comment:
In other places, we take a default `t_ref` if unspecified. https://github.com/gammapy/gammapy/blob/5774c346f72c035b969ecdfe726ede9b68aaf0bf/gammapy/utils/time.py#L21
Should we keep the same behaviour here?